src/cmps/AppStateManager.tsx

```diff
@@ -1,18 +1,53 @@
-import { useState } from "react";
-import Header from "./Header";
-import PokemonContent from "./PokemonContent";
+import { BrowserRouter as Router, Route, Routes } from "react-router-dom";
 import GlobalStyle from "../assets/globalStyles/GlobalStyle";
+import Header from "../cmps/Header";
+import PokemonContent from "../cmps/PokemonContent";
+import FightArenaPage from "../pages/FightArenaPage";
+import { useEffect, useState } from "react";
+import { Pokemon } from "../interfaces/Pokemon";
 
-const AppStateManager = () => {
+function AppStateManager() {
   const [selectedCtg, setSelectedCtg] = useState(0);
+  const [pokemonData, setPokemonData] = useState<Pokemon[]>([]);
+
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+  const fetchData = async () => {
+    const response = await fetch("/src/data/pokemon.json");
+    const data = await response.json();
+    setPokemonData(data);
+  };
+
+  const userPokemons = pokemonData.filter((pokemon) => pokemon.belongsToUser);
```

Comment:
You should use the memo hook, if the pokemonData changed the userPokemons won't update